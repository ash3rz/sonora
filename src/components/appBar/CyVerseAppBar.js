/**
 *
 * @author sriram
 * A component that displays app level tool bar with search and user menu.
 *
 */

import React from "react";
import intlData from "./messages";
import ids from "./ids";
import constants from "../../constants";

import {
    build,
    formatHTMLMessage,
    formatMessage,
    withI18N,
} from "@cyverse-de/ui-lib";

import { useRouter } from "next/router";
import { injectIntl } from "react-intl";
import {
    AppBar,
    ButtonGroup,
    Button,
    ClickAwayListener,
    Grow,
    MenuList,
    MenuItem,
    Paper,
    Popper,
    IconButton,
    InputBase,
    Toolbar,
    Typography,
} from "@material-ui/core";
import { fade, makeStyles, useTheme } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";
import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
import AccountCircleIcon from "@material-ui/icons/AccountCircle";

const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
    },
    appBar: {
        backgroundColor: theme.palette.primary,
    },
    title: {
        flexGrow: 1,
        display: "none",
        [theme.breakpoints.up("sm")]: {
            display: "block",
        },
    },
    search: {
        position: "relative",
        borderRadius: theme.shape.borderRadius,
        backgroundColor: fade(theme.palette.common.white, 0.15),
        "&:hover": {
            backgroundColor: fade(theme.palette.common.white, 0.25),
        },
        marginLeft: 0,
        width: "100%",
        [theme.breakpoints.up("sm")]: {
            marginLeft: theme.spacing(1),
            width: "auto",
        },
    },
    searchIcon: {
        width: theme.spacing(7),
        height: "100%",
        position: "absolute",
        pointerEvents: "none",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
    },
    user: {
        marginLeft: theme.spacing(1),
    },
    inputRoot: {
        color: "inherit",
    },
    inputInput: {
        padding: theme.spacing(1, 1, 1, 7),
        transition: theme.transitions.create("width"),
        width: "100%",
        [theme.breakpoints.up("sm")]: {
            width: 120,
            "&:focus": {
                width: 200,
            },
        },
    },
}));
const searchOptions = ["All", "Data", "Apps", "Analyses"];

function CyverseAppBar(props) {
    const theme = useTheme();
    const classes = useStyles();
    const router = useRouter();
    const { intl, children } = props;

    const handleUserButtonClick = (event) => {
        const { user } = props.children.props;
        if (!user) {
            router.push("/login");
        }
    };

    return (
        <React.Fragment>
            <div className={classes.root}>
                <AppBar position="static" className={classes.appBar}>
                    <Toolbar>
                        <a
                            href={constants.CYVERSE_URL}
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            <img
                                src="/cyverse_whitelogo.png"
                                alt="CyVerse"
                            ></img>
                        </a>
                        <Typography
                            className={classes.title}
                            variant="subtitle2"
                            noWrap
                        >
                            {formatHTMLMessage("discovery")}
                            <br />
                            {formatHTMLMessage("environment")}
                        </Typography>

                        <div
                            className={classes.search}
                            id={build(ids.APP_BAR_BASE_ID, ids.SEARCH)}
                        >
                            <div className={classes.searchIcon}>
                                <SearchIcon />
                            </div>
                            <InputBase
                                placeholder={formatMessage(intl, "search")}
                                classes={{
                                    root: classes.inputRoot,
                                    input: classes.inputInput,
                                }}
                                inputProps={{
                                    "aria-label": formatMessage(
                                        intl,
                                        "searchAriaLabel"
                                    ),
                                }}
                            />
                        </div>
                        <SearchOptions intl={intl} />
                        <div className={classes.user}>
                            <IconButton
                                id={build(
                                    ids.APP_BAR_BASE_ID,
                                    ids.USER_MENU_BTN
                                )}
                                aria-label={formatMessage(
                                    intl,
                                    "userMenuAriaLabel"
                                )}
                                onClick={(event) => {
                                    handleUserButtonClick(event);
                                }}
                            >
                                <AccountCircleIcon
                                    fontSize="large"
                                    style={{
                                        color:
                                            theme.palette.primary.contrastText,
                                    }}
                                />
                            </IconButton>
                        </div>
                    </Toolbar>
                </AppBar>
            </div>
            {children}
        </React.Fragment>
    );
}

function SearchOptions(props) {
    const { intl } = props;
    const theme = useTheme();
    const [open, setOpen] = React.useState(false);
    const anchorRef = React.useRef(null);
    const [selectedIndex, setSelectedIndex] = React.useState(0);

    const handleClick = () => {
        //as of now do nothing
        console.info(`You clicked ${searchOptions[selectedIndex]}`);
    };

    const handleMenuItemClick = (event, index) => {
        setSelectedIndex(index);
        setOpen(false);
    };

    const handleToggle = () => {
        setOpen((prevOpen) => !prevOpen);
    };

    const handleClose = (event) => {
        if (anchorRef.current && anchorRef.current.contains(event.target)) {
            return;
        }

        setOpen(false);
    };
    return (
        <Paper>
            <ButtonGroup
                variant="contained"
                ref={anchorRef}
                aria-label={formatMessage(intl, "searchFilterAriaLabel")}
            >
                <Button
                    id={build(ids.APP_BAR_BASE_ID, ids.SEARCH_FILTER_BTN)}
                    onClick={handleClick}
                    style={{ backgroundColor: theme.palette.white }}
                >
                    {searchOptions[selectedIndex]}
                </Button>
                <Button
                    id={build(
                        ids.APP_BAR_BASE_ID,
                        ids.SEARCH_FILTER_OPTIONS_BTN
                    )}
                    style={{ backgroundColor: theme.palette.white }}
                    size="small"
                    aria-controls={
                        open
                            ? formatMessage(intl, "searchFilterAriaControl")
                            : undefined
                    }
                    aria-expanded={open ? "true" : undefined}
                    aria-label={formatMessage(
                        intl,
                        "searchFilterMenuAriaLabel"
                    )}
                    aria-haspopup="menu"
                    onClick={handleToggle}
                >
                    <ArrowDropDownIcon />
                </Button>
            </ButtonGroup>
            <Popper
                open={open}
                anchorEl={anchorRef.current}
                role={undefined}
                transition
                disablePortal
            >
                {({ TransitionProps, placement }) => (
                    <Grow
                        {...TransitionProps}
                        style={{
                            transformOrigin:
                                placement === "bottom"
                                    ? "center top"
                                    : "center bottom",
                        }}
                    >
                        <Paper>
                            <ClickAwayListener onClickAway={handleClose}>
                                <MenuList
                                    id={build(
                                        ids.APP_BAR_BASE_ID,
                                        ids.SEARCH_FILTER_MENU
                                    )}
                                >
                                    {searchOptions.map((option, index) => (
                                        <MenuItem
                                            key={option}
                                            id={build(
                                                ids.APP_BAR_BASE_ID,
                                                ids.SEARCH_FILTER_MENU +
                                                    "." +
                                                    option
                                            )}
                                            selected={index === selectedIndex}
                                            onClick={(event) =>
                                                handleMenuItemClick(
                                                    event,
                                                    index
                                                )
                                            }
                                        >
                                            {option}
                                        </MenuItem>
                                    ))}
                                </MenuList>
                            </ClickAwayListener>
                        </Paper>
                    </Grow>
                )}
            </Popper>
        </Paper>
    );
}

export default withI18N(injectIntl(CyverseAppBar), intlData);
