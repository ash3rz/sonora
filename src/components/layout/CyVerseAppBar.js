/**
 *
 * @author sriram
 * A component that displays app level tool bar with search and user menu. Also renders a drawer menu
 * in small screens instead of Navigation bar
 *
 */

import React, { useEffect, useRef, useState } from "react";

import { useQuery } from "react-query";
import { useRouter } from "next/router";
import { injectIntl } from "react-intl";

import ids from "./ids";
import intlData from "./messages";
import constants from "../../constants";
import GlobalSearchField from "../search/GlobalSearchField";
import NavigationConstants from "../../common/NavigationConstants";
import Notifications from "./Notifications";
import CustomIntercom from "./CustomIntercom";
import { useUserProfile } from "../../contexts/userProfile";
import { getUserProfile } from "../../serviceFacades/users";

import {
    build,
    CyVerseAnnouncer,
    formatHTMLMessage,
    formatMessage,
    getMessage,
    withI18N,
} from "@cyverse-de/ui-lib";

import {
    AppBar,
    Avatar,
    Divider,
    Drawer,
    Hidden,
    IconButton,
    List,
    ListItem,
    ListItemIcon,
    ListItemText,
    Toolbar,
    Typography,
} from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";
import AccountCircle from "@material-ui/icons/AccountCircle";
import MenuIcon from "@material-ui/icons/Menu";
import SettingsIcon from "@material-ui/icons/Settings";

const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
    },
    appBar: {
        backgroundColor: theme.palette.bgGray,
        boxShadow: 0,
    },
    drawerIcon: {
        height: 18,
        width: 18,
        paddingRight: theme.spacing(4),
    },
    margin: {
        [theme.breakpoints.up("sm")]: {
            margin: theme.spacing(2),
        },
        [theme.breakpoints.down("xs")]: {
            margin: theme.spacing(1),
        },
    },
    userIcon: {
        cursor: "pointer",
        margin: theme.spacing(2),
        backgroundColor: theme.palette.success.main,
        color: theme.palette.success.contrastText,
    },
}));

function CyverseAppBar(props) {
    const classes = useStyles();
    const router = useRouter();
    const ref = useRef();

    const { intl, children, setAppBarRef } = props;
    const [userProfile, setUserProfile] = useUserProfile();
    const [avatarLetter, setAvatarLetter] = useState("");

    const [drawerOpen, setDrawerOpen] = useState(false);
    useQuery({
        queryKey: "getUserProfile",
        queryFn: getUserProfile,
        config: {
            onSuccess: setUserProfile,
        },
    });

    useEffect(() => {
        setAppBarRef(ref);
    }, [ref, setAppBarRef]);

    React.useEffect(() => {
        if (userProfile?.id) {
            setAvatarLetter(userProfile.id.charAt(0).toUpperCase());
        }
    }, [userProfile, setAvatarLetter]);

    const handleUserButtonClick = (event) => {
        if (!userProfile) {
            router.push(`/${NavigationConstants.LOGIN}${router.asPath}`);
        } else {
            router.push(`/${NavigationConstants.LOGOUT}`);
        }
    };
    const handleSearchClick = (event) => {
        router.push("/" + NavigationConstants.SEARCH);
        toggleDrawer(false);
    };
    const toggleDrawer = (open) => (event) => {
        setDrawerOpen(open);
    };

    return (
        <>
            <div className={classes.root}>
                <AppBar
                    id={ids.APP_BAR_BASE}
                    position="static"
                    variant="outlined"
                    className={classes.appBar}
                    ref={ref}
                >
                    <Toolbar>
                        <Hidden only={["sm", "md", "lg", "xl"]}>
                            <div>
                                <IconButton
                                    id={build(
                                        ids.APP_BAR_BASE,
                                        ids.DRAWER_MENU_BTN
                                    )}
                                    edge="start"
                                    className={classes.menuButton}
                                    color="primary"
                                    aria-label={formatMessage(
                                        intl,
                                        "drawerMenuAriaLabel"
                                    )}
                                    onClick={toggleDrawer(true)}
                                >
                                    <MenuIcon />
                                </IconButton>
                            </div>
                            <Drawer
                                id={ids.DRAWER_MENU}
                                open={drawerOpen}
                                onClose={toggleDrawer(false)}
                            >
                                <List onClick={toggleDrawer(false)}>
                                    <ListItem
                                        id={build(
                                            ids.DRAWER_MENU,
                                            ids.ACCOUNT_MI
                                        )}
                                        onClick={handleUserButtonClick}
                                    >
                                        {userProfile ? (
                                            <>
                                                <ListItemIcon>
                                                    <Avatar
                                                        className={
                                                            classes.userIcon
                                                        }
                                                    >
                                                        <Typography
                                                            variant={"h6"}
                                                        >
                                                            {avatarLetter}
                                                        </Typography>
                                                    </Avatar>
                                                </ListItemIcon>
                                                <ListItemText>
                                                    {getMessage("logout")}
                                                </ListItemText>
                                            </>
                                        ) : (
                                            <>
                                                <ListItemIcon>
                                                    <AccountCircle />
                                                </ListItemIcon>
                                                <ListItemText>
                                                    {getMessage("login")}
                                                </ListItemText>
                                            </>
                                        )}
                                    </ListItem>
                                    <Divider />
                                    <ListItem
                                        id={build(
                                            ids.DRAWER_MENU,
                                            ids.DASHBOARD_MI
                                        )}
                                        onClick={() =>
                                            router.push(
                                                "/" +
                                                    NavigationConstants.DASHBOARD
                                            )
                                        }
                                    >
                                        <img
                                            className={classes.drawerIcon}
                                            src="/dashboard.png"
                                            alt={formatMessage(
                                                intl,
                                                "dashboard"
                                            )}
                                        />
                                        <ListItemText>
                                            {getMessage("dashboard")}
                                        </ListItemText>
                                    </ListItem>
                                    <ListItem
                                        id={build(ids.DRAWER_MENU, ids.DATA_MI)}
                                        onClick={() =>
                                            router.push(
                                                "/" + NavigationConstants.DATA
                                            )
                                        }
                                    >
                                        <img
                                            className={classes.drawerIcon}
                                            src="/data.png"
                                            alt={formatMessage(
                                                intl,
                                                "dashboard"
                                            )}
                                        />
                                        <ListItemText>
                                            {getMessage("data")}
                                        </ListItemText>
                                    </ListItem>
                                    <ListItem
                                        id={build(ids.DRAWER_MENU, ids.APPS_MI)}
                                        onClick={() =>
                                            router.push(
                                                "/" + NavigationConstants.APPS
                                            )
                                        }
                                    >
                                        <img
                                            className={classes.drawerIcon}
                                            src="/apps.png"
                                            alt={formatMessage(intl, "apps")}
                                        />
                                        <ListItemText>
                                            {getMessage("apps")}
                                        </ListItemText>
                                    </ListItem>
                                    <ListItem
                                        id={build(
                                            ids.DRAWER_MENU,
                                            ids.ANALYSES_MI
                                        )}
                                        onClick={() =>
                                            router.push(
                                                "/" +
                                                    NavigationConstants.ANALYSES
                                            )
                                        }
                                    >
                                        <img
                                            className={classes.drawerIcon}
                                            src="/analyses.png"
                                            alt={formatMessage(
                                                intl,
                                                "analyses"
                                            )}
                                        />
                                        <ListItemText>
                                            {getMessage("analyses")}
                                        </ListItemText>
                                    </ListItem>
                                    <ListItem
                                        id={build(
                                            ids.DRAWER_MENU,
                                            ids.SEARCH_MI
                                        )}
                                        onClick={handleSearchClick}
                                    >
                                        <ListItemIcon>
                                            <SearchIcon />
                                        </ListItemIcon>
                                        <ListItemText>
                                            {getMessage("search")}
                                        </ListItemText>
                                    </ListItem>
                                    <Divider />
                                    <ListItem
                                        id={build(
                                            ids.DRAWER_MENU,
                                            ids.SETTINGS_MI
                                        )}
                                        onClick={() =>
                                            router.push(
                                                "/" +
                                                    NavigationConstants.SETTINGS
                                            )
                                        }
                                    >
                                        <ListItemIcon>
                                            <SettingsIcon />
                                        </ListItemIcon>
                                        <ListItemText>
                                            {getMessage("settings")}
                                        </ListItemText>
                                    </ListItem>
                                </List>
                            </Drawer>
                            <Typography color="primary">
                                {formatHTMLMessage("discovery")}
                                &nbsp;{formatHTMLMessage("environment")}
                            </Typography>
                        </Hidden>
                        <Hidden xsDown>
                            <a
                                href={constants.CYVERSE_URL}
                                target="_blank"
                                rel="noopener noreferrer"
                            >
                                <img
                                    src="/de.png"
                                    alt={formatMessage(intl, "cyverse")}
                                ></img>
                            </a>
                        </Hidden>
                        <Hidden smDown>
                            <GlobalSearchField />
                        </Hidden>
                        <div className={classes.root} />
                        <div style={{ display: "flex" }}>
                            <CustomIntercom classes={classes} intl={intl} />
                            <Notifications intl={intl} classes={classes} />
                            <Hidden only={["xs", "md", "lg", "xl"]}>
                                <IconButton
                                    id={build(ids.APP_BAR_BASE, ids.SEARCH_BTN)}
                                    aria-label={formatMessage(
                                        intl,
                                        "searchAriaLabel"
                                    )}
                                    aria-controls={formatMessage(
                                        intl,
                                        "searchButtonAriaControl"
                                    )}
                                    color="primary"
                                    onClick={handleSearchClick}
                                    size="small"
                                    className={classes.margin}
                                >
                                    <SearchIcon />
                                </IconButton>
                            </Hidden>
                            <Hidden xsDown>
                                {userProfile ? (
                                    <Avatar
                                        onClick={handleUserButtonClick}
                                        id={build(
                                            ids.APP_BAR_BASE,
                                            ids.ACCOUNT_BTN
                                        )}
                                        className={classes.userIcon}
                                    >
                                        <Typography variant={"h6"}>
                                            {avatarLetter}
                                        </Typography>
                                    </Avatar>
                                ) : (
                                    <IconButton
                                        id={build(
                                            ids.APP_BAR_BASE,
                                            ids.ACCOUNT_BTN
                                        )}
                                        aria-label={formatMessage(
                                            intl,
                                            "accountAriaLabel"
                                        )}
                                        aria-controls={formatMessage(
                                            intl,
                                            "accountAriaControl"
                                        )}
                                        color="primary"
                                        onClick={handleUserButtonClick}
                                        size="small"
                                        className={classes.margin}
                                    >
                                        <AccountCircle />
                                    </IconButton>
                                )}
                            </Hidden>
                        </div>
                    </Toolbar>
                </AppBar>
                <CyVerseAnnouncer />
            </div>
            {children}
        </>
    );
}

export default withI18N(injectIntl(CyverseAppBar), intlData);
