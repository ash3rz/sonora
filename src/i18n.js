import NextI18Next from "next-i18next";
import path from "path";
import React, { Suspense } from "react";
import { I18nextProvider } from "react-i18next";

/**
 * The singular instance of next-i18next to be used throughout the app.
 * Every page and component that needs i18n should use appWithTranslation,
 * useTranslation, and withTranslation exported from this instance
 *
 * See examples here from the next-i18next repo:
 * https://github.com/isaachinman/next-i18next/tree/master/examples/simple
 */
const {
    i18n,
    useTranslation,
    withTranslation,
    appWithTranslation,
} = new NextI18Next({
    defaultLanguage: "en",
    otherLanguages: [],
    localePath: path.resolve("./public/static/locales"),
});

/**
 * A wrapper component to provide i18n capabilities to Storybook as well as
 * jest tests
 *
 * Copied suggestion from https://github.com/isaachinman/next-i18next/issues/324
 *
 * @param children
 * @returns {JSX.Element}
 * @constructor
 */
function I18nProviderWrapper({ children }) {
    return (
        <Suspense fallback={"Loading..."}>
            <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
        </Suspense>
    );
}

// reexport everything
export {
    i18n,
    useTranslation,
    withTranslation,
    appWithTranslation,
    I18nProviderWrapper,
};
