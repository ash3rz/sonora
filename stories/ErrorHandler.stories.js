import React from "react";
import { select, withKnobs } from "@storybook/addon-knobs";
import ErrorHandler from "../src/components/utils/error/ErrorHandler";

const errorResponse = {
    data: {
        error_code: "ERR_ILLEGAL_ARGUMENT",
        reason: {
            sortfield: "disallowed-key",
        },
    },
    status: 400,
    statusText: "Bad Request",
    headers: {
        connection: "keep-alive",
        "content-length": "77",
        "content-type": "application/json; charset=utf-8",
        date: "Tue, 21 Apr 2020 08:51:22 GMT",
        etag: 'W/"4d-DGj9xs+ftEbvtNhFlVhDEbAXxQo"',
        "x-powered-by": "Express",
    },
    config: {
        url:
            '/api/analyses?limit=25&sortfield=startdate&sort-dir=DESC&offset=0&filter=[{"field":"parent_id","value":""},{"field":"ownership","value":"all"}]',
        method: "get",
        headers: {
            Accept: "application/json, text/plain, *!/!*",
        },
        transformRequest: [null],
        transformResponse: [null],
        timeout: 20000,
        xsrfCookieName: "XSRF-TOKEN",
        xsrfHeaderName: "X-XSRF-TOKEN",
        maxContentLength: -1,
        credentials: "include",
    },
    request: {},
};

const errorObject = {
    message: "Request failed with status code 400",
    name: "Error",
    stack:
        "Error: Request failed with status code 400\n    at createError (http://localhost:3000/_next/static/development/pages/analyses.js?ts=1587458580850:203736:15)\n    at settle (http://localhost:3000/_next/static/development/pages/analyses.js?ts=1587458580850:203997:12)\n    at XMLHttpRequest.handleLoad (http://localhost:3000/_next/static/development/pages/analyses.js?ts=1587458580850:203205:7)",
    config: {
        url:
            '/api/analyses?limit=25&sortfield=startdate&sort-dir=DESC&offset=0&filter=[{"field":"parent_id","value":""},{"field":"ownership","value":"all"}]',
        method: "get",
        headers: {
            Accept: "application/json, text/plain, *!/!*",
        },
        transformRequest: [null],
        transformResponse: [null],
        timeout: 20000,
        xsrfCookieName: "XSRF-TOKEN",
        xsrfHeaderName: "X-XSRF-TOKEN",
        maxContentLength: -1,
        credentials: "include",
    },
    response: errorResponse,
};

const loginErrorResponse = {
    data: {
        error_code: "ERR_NOT_AUTHORIZED",
        reason: "No authentication information found in request.",
    },
    status: 401,
    statusText: "Internal Server Error",
    headers: {
        connection: "keep-alive",
        "content-length": "94",
        "content-type": "application/json; charset=utf-8",
        date: "Tue, 21 Apr 2020 14:05:44 GMT",
        etag: 'W/"5e-9SP9RtM3QQRwZ3UNFelGCk5ZHaI"',
        "x-powered-by": "Express",
    },
    config: {
        url: "/api/apps/categories?public=false",
        method: "get",
        headers: { Accept: "application/json, text/plain, */*" },
        transformRequest: [null],
        transformResponse: [null],
        timeout: 20000,
        xsrfCookieName: "XSRF-TOKEN",
        xsrfHeaderName: "X-XSRF-TOKEN",
        maxContentLength: -1,
        credentials: "include",
    },
    request: {},
};

const loginErrorObject = {
    message: "Request failed with status code 401",
    name: "Error",
    stack:
        "Error: Request failed with status code 401\n    at createError (http://localhost:3000/_next/static/development/pages/_app.js?ts=1587477886272:196870:15)\n    at settle (http://localhost:3000/_next/static/development/pages/_app.js?ts=1587477886272:197131:12)\n    at XMLHttpRequest.handleLoad (http://localhost:3000/_next/static/development/pages/_app.js?ts=1587477886272:196339:7)",
    config: {
        url: "/api/apps/categories?public=false",
        method: "get",
        headers: { Accept: "application/json, text/plain, */*" },
        transformRequest: [null],
        transformResponse: [null],
        timeout: 20000,
        xsrfCookieName: "XSRF-TOKEN",
        xsrfHeaderName: "X-XSRF-TOKEN",
        maxContentLength: -1,
        credentials: "include",
    },
    response: loginErrorResponse,
};

export default { title: "Error Handler", decorators: [withKnobs] };

export function ErrorHandlerTest() {
    const label = "Type";
    const options = ["empty", "401", "500"];
    const defaultValue = "empty";
    const groupId = "errors";
    const value = select(label, options, defaultValue, groupId);
    let errObj = null;
    if (value === "401") {
        errObj = loginErrorObject;
    } else if (value === "500") {
        errObj = errorObject;
    }

    return <ErrorHandler errorObject={errObj} baseId="errorBase" />;
}
