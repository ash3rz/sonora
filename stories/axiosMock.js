/**
 * @author sriram
 *
 * Create and export a mock axios object to be shared by all tests.
 *
 */
import axiosInstance from "../src/common/getAxios";
import MockAdapter from "axios-mock-adapter";
const mockAxios = new MockAdapter(axiosInstance, { delayResponse: 2000 });
export { mockAxios };
