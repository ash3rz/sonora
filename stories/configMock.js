export default {
    intercom: {
        appId: "appId",
        enabled: true,
        companyId: "companyId",
        companyName: "companyName",
    },
    sessions: {
        poll_interval_ms: 5000,
    },
    admin: {
        groups: "test",
        group_attribute_name: "entitlement",
    },
    analysis: { supportUser: { user: "support_user", source_id: "ldap" } },
    irods: {
        home_path: "/iplant/home",
        trash_path: "/iplant/trash/home/de-irods",
    },
    vice: {
        initContainerName: "input-files-init",
        inputFilesContainerName: "input-files",
        viceProxyContainerName: "vice-proxy",
        analysisContainerName: "analysis",
    },
};
