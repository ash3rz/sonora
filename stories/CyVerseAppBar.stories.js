import React from "react";
import { UserProfileProvider } from "../src/contexts/userProfile";
import { ConfigProvider } from "../src/contexts/config";
import CyverseAppBar from "../src/components/layout/CyVerseAppBar";
import { mockAxios } from "./axiosMock";
import { NotificationsProvider } from "../src/contexts/pushNotifications";
import testConfig from "./configMock";

const mockUser = {
    id: "mockUser",
    attributes: {
        credentialType: "UsernamePasswordCredential",
        email: "mockuser@example.org",
        entitlement: ["foo", "bar", "baz"],
        firstName: "Mock",
        lastName: "User",
        name: "Mock User",
    },
};
function AppBarTest() {
    mockAxios.onGet("/api/profile").reply(200, mockUser);
    return (
        <UserProfileProvider>
            <NotificationsProvider>
                <ConfigProvider>
                    <CyverseAppBar
                        setAppBarRef={() => {}}
                        clientConfig={testConfig}
                    />
                </ConfigProvider>
            </NotificationsProvider>
        </UserProfileProvider>
    );
}

export default { title: "App Bar" };

export const AppBar = () => {
    return <AppBarTest />;
};
