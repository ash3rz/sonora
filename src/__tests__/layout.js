import React from "react";
import renderer from "react-test-renderer";
import { AppBar } from "../../stories/CyVerseAppBar.stories";
import { I18nProviderWrapper } from "../i18n";

test("App Bar renders", () => {
    const component = renderer.create(
        <I18nProviderWrapper>
            <AppBar />
        </I18nProviderWrapper>
    );
    component.unmount();
});
