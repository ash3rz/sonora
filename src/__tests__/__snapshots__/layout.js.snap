// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Bar renders 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-2 MuiPaper-outlined"
    id="appBar"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        className="makeStyles-root-1"
      />
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <button
          aria-label="show new notifications"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-notification-4 MuiIconButton-colorPrimary"
          disabled={false}
          id="appBar.notificationButton"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorError MuiBadge-invisible"
              >
                0
              </span>
            </span>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;

exports[`Nav Bar renders 1`] = `null`;
